src/printer.js

```diff
@@ -2802,6 +2790,34 @@ function printBinaryishExpressions(path, parts, print, options, isNested) {
   return parts;
 }
 
+function printAssignment(printedLeft, operator, rightNode, printedRight, options) {
```

Comment:
We copy and pasted this non trivial logic twice. I extracted it into its own function